tests/build/test_build.py

```
@@ -0,0 +1,80 @@
+        # Load environment variables
+        load_dotenv()
+        test_port = os.getenv("TEST_PORT", "8502")
+        
+        # Get the project root (two levels up from tests/build/test_build.py)
+        project_root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+        
+        # Correct path to main.py in the src directory
+        app_path = os.path.join(project_root, "src", "main.py")
+
+        # Start Streamlit server in background
+        process = subprocess.Popen(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)